pkg/kev/converter/kubernetes/transform.go

```diff
@@ -645,39 +647,26 @@ func (k *Kubernetes) initIngress(projectService ProjectService, port int32) *net
 			APIVersion: "networking.k8s.io/v1beta1",
 		},
 		ObjectMeta: meta.ObjectMeta{
-			Name:   projectService.Name,
-			Labels: configLabels(projectService.Name),
-			// Annotations: configAnnotations(projectService),
+			Name:        projectService.Name,
+			Labels:      configLabels(projectService.Name),
 			Annotations: projectService.ingressAnnotations(),
 		},
-		Spec: networkingv1beta1.IngressSpec{
-			Rules: make([]networkingv1beta1.IngressRule, len(hosts)),
-		},
+		Spec: networkingv1beta1.IngressSpec{},
 	}
 
-	for i, host := range hosts {
-		host, p := parseIngressPath(host)
-		ingress.Spec.Rules[i] = networkingv1beta1.IngressRule{
-			IngressRuleValue: networkingv1beta1.IngressRuleValue{
-				HTTP: &networkingv1beta1.HTTPIngressRuleValue{
-					Paths: []networkingv1beta1.HTTPIngressPath{
-						{
-							Path: p,
-							Backend: networkingv1beta1.IngressBackend{
-								ServiceName: projectService.Name,
-								ServicePort: intstr.IntOrString{
-									IntVal: port,
-								},
-							},
-						},
-					},
-				},
-			},
-		}
-		if host != "true" {
-			ingress.Spec.Rules[i].Host = host
+	if hasDefaultIngressBackendKeyword(hosts) {
+		ingress.Spec.Rules = []networkingv1beta1.IngressRule{
+			createIngressRule("", "", projectService.Name, port),
```

Comment:
I wonder whether this is going to result with a valid ingress path? Perhaps "/" as path would be more appropriate? Worth double checking with the docs for this one.

Reply:
Thanks for the catch! And, **you shouldn't need a path**. In fact the implementation was wrong.

The default backend should not produce any rules. It should create an ingress object similar to this:

```yaml
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: test-ingress
spec:
  defaultBackend:
    service:
      name: test
      port:
        number: 80
```

I've corrected the implementation to do the above using the `*networkingv1beta1.IngressBackend` struct.